src/search.c

```diff
@@ -457,6 +457,9 @@ int search(Thread *thread, PVariation *pv, int alpha, int beta, int depth) {
         if (!apply(thread, board, move))
             continue;
 
+        // If the king is now attacked, then this move gave check
+        gaveCheck = !!board->kingAttackers;
+
```

Comment:
I don't think we care to set a variable. ``board->kingAttackers`` will still be set when used below, and your comments make it clear what the intention is. The declaration of ``gaveCheck`` as well.